Makefile

```diff
@@ -204,6 +204,10 @@ ifneq ($(MM_NO_ENTERPRISE_LINT),true)
 endif
 endif
 
+app-layers: ## Extract interface from App struct
```

Comment:
While this is cool for a one time use, I don't think we should encourage this. 
I would much rather break up app into services.

Reply:
"break up app into services" - can you elaborate please?

Reply:
Instead of having all the business logic in app refactor that stuff out. Like the enterprise code and all the packages in the `services` folder.

Reply:
@crspeller - ah. Well, this is definitely planned for the future, but at the moment - we need to abstract App into an interface to allow integration of OpenTracing - that's what this PR is for